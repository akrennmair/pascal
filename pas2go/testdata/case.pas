program test;

label 10;

var i : integer;

begin
	i := 4;
	10:
	case i of
	0: writeln('goodbye world');
	1: writeln('hello world');
	2, 3, 4:
		begin
			writeln('wait for it...');
			i := i - 1;
			goto 10
		end;
	end
end.
