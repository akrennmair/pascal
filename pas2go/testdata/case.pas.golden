package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		i int
	)
	_ = i

	i = 4
L10:
	switch i {
	case 0:
		system.Writeln("goodbye world")
	case 1:
		system.Writeln("hello world")
	case 2, 3, 4:
		system.Writeln("wait for it...")
		i = i - 1
		goto L10
	}
}
